lib/core/src/sdk.rs

```diff
@@ -1431,6 +1440,118 @@ impl LiquidSdk {
             }
         }
 
+        // TODO Ensure this happens only on first sync
+        // TODO This must be done after:
+        //  1) we fetched LWK onchain data, so that we have tx list and LWK is initialized
+        //  2) we fetched real-time sync data
+
+        let con = self.persister.get_connection()?;
+        let t0 = Instant::now();
+        // Send Swap scripts by swap ID
+        let send_swap_immutable_db: HashMap<String, LBtcSwapScriptV2> = self
+            .persister
+            .list_send_swaps(&con, vec![])?
+            .iter()
+            .filter_map(|swap| match swap.get_swap_script() {
+                Ok(script) => Some((swap.id.clone(), script)),
+                Err(e) => {
+                    error!("Failed to get swap script for Send Swap {}: {e}", swap.id);
+                    None
+                }
+            })
+            .collect();
+        let send_swap_immutable_db_size = send_swap_immutable_db.len();
+        info!("Send Swap immutable DB: {send_swap_immutable_db_size} rows");
+
+        self.recover_send_swap_tx_ids(tx_map, send_swap_immutable_db)
+            .await?;
+
+        let duration_ms = Instant::now().duration_since(t0).as_millis();
+        info!("Finished recovering swap tx IDs from onchain data (t = {duration_ms} ms)");
+
+        Ok(())
+    }
+
+    /// Reconstruct Send Swap tx IDs from the onchain data and the immutable DB data
+    async fn recover_send_swap_tx_ids(
+        &self,
+        tx_map: HashMap<String, WalletTx>,
+        send_swap_immutable_db_data: HashMap<String, LBtcSwapScriptV2>,
+    ) -> Result<()> {
+        // Find incoming txs that have a single input and a single output, indexed by input
+        let possible_claim_or_refund_txs_by_input: HashMap<OutPoint, Txid> = tx_map
+            .iter()
+            .filter(|(_, tx)| tx.balance.values().sum::<i64>() > 0)
+            .filter(|(_, tx)| tx.tx.output.len() == 2) // Refund tx + fee
+            .filter(|(_, tx)| tx.tx.input.len() == 1)
+            .filter_map(|(_, wallet_tx)| {
+                wallet_tx
+                    .tx
+                    .input
+                    .first()
```

Comment:
I agree it will work but if on future versions we will find a way to aggregate multiple refunds in one transaction (to save fees) it will break. So perhaps just map all inputs?